### PR TITLE
M14

### PR DESCRIPTION
**Pull request by Dani2525**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -86,5 +86,43 @@ ARM_SUBSCRIPTION_ID
...
+kubectl apply -f deployment.yaml  
+kubectl apply -f service.yaml
+kubectl apply -f secrets.yaml
+```
+
+## forward to port    
+```bash 
+kubectl port-forward service/module-14 5000:5000
+```
+it is now running from local minikibe cluster and can be accessed.
```
Comment (JackMeadDev) [inline README.md:139]: Thanks for the nice clear instructions!

File: deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            - name: LOGGLY_TOKEN
+              valueFrom:
+                secretKeyRef:
+                 name: secret
+                 key: LOGGLY_TOKEN
+            - name: LOG_LEVEL
+              valueFrom:
+                secretKeyRef:
+                 name: secret
+                 key: LOG_LEVEL
```
Comment (JackMeadDev) [inline deployment.yaml:49]: This value isn't sensitive, so we could set it directly, like:
```suggestion
            - name: LOG_LEVEL
              value: INFO
```

File: secrets.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+metadata:
+  name: secret
+type: Opaque
+data:
+  secret_key:  TVlfU0VDUkVUX0tFWQ== # is encoded version of MY_SECRET_KEY
+  client_id:  NDYxNjQwZmQxMDAwNjQyZDA0NjI=
+  client_secret:  YTFhMmU4ZjFkYzMxMDM1MWQxOGI0YTNhMzY0OTNjN2VkYzljMGUyMg==
+  mongo_client:  bW9uZ29kYjovL21vZHVsZTEwOnQwcXd4VTNWN01aMlk1NW1oYm5SQXRlZmlHcndKVFVIYVRPNk1oTzBLdjhLVFczUFg2R0t2bG9uM0xGMWg4dTI5QkNnUVJZRTQxd2dBQ0RiU1R6c2dnPT1AbW9kdWxlMTAubW9uZ28uY29zbW9zLmF6dXJlLmNvbToxMDI1NS8/c3NsPXRydWUmcmVwbGljYVNldD1nbG9iYWxkYiZyZXRyeXdyaXRlcz1mYWxzZSZtYXhJZGxlVGltZU1TPTEyMDAwMCZhcHBOYW1lPUBtb2R1bGUxMEA=
+  LOGGLY_TOKEN:  NTI4NmEwNGYtMTZhYy00ZjA1LTgzN2YtNGY3MTJkOWM4MDk1
+  LOG_LEVEL:  REVCVUc=
```
Comment (JackMeadDev) [inline secrets.yaml]: These are only encoded, not encrypted, so they're now all public knowledge!

We should remove this file (add it to .gitignore, and remove the existing memory with `git rm --cached secrets.yaml`), although it is nice to leave a template version that doesn't have the secrets, i.e. we could have a new file called `secrets.yaml.template` that just has:
```
apiVersion: v1
kind: Secret
metadata:
  name: secret
type: Opaque
data:
  secret_key:  
  client_id:  
  client_secret:  
  mongo_client: 
  LOGGLY_TOKEN:  
  LOG_LEVEL:  
```

**Review by JackMeadDev — APPROVED** [review]
Nice work, I was easily able to follow your instructions and set up your app within Minikube 😄 I've left a couple of optional thoughts below, but the key one is that we don't want your secrets checked in here - we should now treat those as compromised and remove them (more info below) but also rotate them (particularly the MongoDB password!). Once that's done, I'm happy for this to be merged.

I'm happy for you to merge once that's done; congratulations on finishing the final exercise! 

By default I'll schedule the resources to be torn down in Azure in two weeks time, but let me know if there's any issues with that 🙂

File: deployment.yaml (hunk 2 above)
Comment (Dani2525) [reply deployment.yaml:49]: done

File: secrets.yaml (hunk 3 above)
Comment (Dani2525) [reply secrets.yaml]: done :)

**Pull request merged**